sdk/evm/tbrv3/contract.ts

```diff
@@ -118,27 +133,79 @@ export interface Transfer {
+  constructor(
+    public readonly provider: ethers.Provider,
+    public readonly network: NetworkMain,
+    public readonly address: string = Tbrv3.addresses[network],
+    private readonly gasToken: string,
+  ) {
     this.address = address;
+
+  }
+
+  static connect(
+    provider: ethers.Provider,
+    network: NetworkMain,
+    chain: Chain,
+    gasToken?: string,
+    address?: string,
+  ) {
+    if (address === undefined && this.addresses[network] === "TBD") {
+      throw new Error(`Tbrv3 address needs to be provided for network ${network}`);
+    }
+
+    const defaultGasToken = this.gasTokens[network][chain];
+    if (gasToken === undefined) {
+      if (defaultGasToken === undefined) {
+        throw new Error(`Gas token address needs to be provided for network ${network} and chain ${chain}`);
+      }
+      gasToken = defaultGasToken;
+    } else if (defaultGasToken !== undefined && gasToken.toLowerCase() !== defaultGasToken?.toLowerCase()) {
+      throw new Error(`Unexpected gas token address ${gasToken} for network ${network} and chain ${chain}`);
+    }
+
+    return new Tbrv3(provider, network, address, gasToken);
+  }
+
+  /**
+   * @deprecated This method will be removed soon. The sdk won't use ethers directly anymore.
+   */
+  static fromRpcUrlStatic(rpc: string, network: NetworkMain, chain: Chain, gasToken?: string, address?: string): Tbrv3 {
+    return this.connect(new ethers.JsonRpcProvider(rpc, undefined, {staticNetwork: true}), network, chain, gasToken, address);
+  }
+
+  /**
+   * @deprecated This method will be removed soon. The sdk won't use ethers directly anymore.
+   */
+  static fromRpcUrl(rpc: string, network: NetworkMain, chain: Chain, gasToken?: string, address?: string): Tbrv3 {
+    return this.connect(new ethers.JsonRpcProvider(rpc), network, chain, gasToken, address);
   }
 
   static readonly addresses: Record<NetworkMain, string> = {
     Mainnet: "TBD",
     Testnet: "TBD",
   }
 
-  transferWithRelay(...transfers: Transfer[]): TbrPartialTx {
+  // TODO: fill these out
+  static readonly gasTokens: Record<NetworkMain, Partial<Record<Chain, string>>> = {
+    Mainnet: {},
+    Testnet: {},
+  }
+
+  transferWithRelay(allowances: TokenBridgeAllowances, ...transfers: Transfer[]): TbrPartialTx {
     if (transfers.length === 0) {
       throw new Error("At least one transfer should be specified.");
     }
 
+    const requiredAllowances: Record<string, bigint> = {};
+    for (const token of Object.keys(allowances)) {
+      requiredAllowances[token] = 0n;
+    }
     // TODO: decide if we want to check that requested gas dropoff doesn't exceed max gas dropoff per transfer
     // Here we need to batch `this.baseRelayingParams` per target chain together with the relaying fee per transfer
 
     let value: bigint = 0n;
-    const methodCalls = [];
+    const transferCalls: LayoutToType<typeof commandCategoryLayout>[] = [];
```

Comment:
```suggestion
    const transferCalls: CommandCategory[] = [];
```
LayoutToType types are exported by layout.ts already.